python/cugraph/cugraph/dask/sampling/uniform_neighbor_sample.py

```diff
@@ -396,23 +396,41 @@ def uniform_neighbor_sample(
     else:
         indices_t = numpy.int32
 
-    if input_graph.renumbered:
-        start_list = input_graph.lookup_internal_vertex_id(start_list)
-
-    start_list = start_list.rename(start_col_name).to_frame()
+    start_list = start_list.rename(start_col_name)
     if batch_id_list is not None:
-        ddf = start_list.join(batch_id_list.rename(batch_col_name))
+        batch_id_list = batch_id_list.rename(batch_col_name)
+        if hasattr(start_list, "compute"):
+            # mg input
+            start_list = start_list.to_frame()
+            batch_id_list = batch_id_list.to_frame()
+            ddf = start_list.merge(
+                batch_id_list,
+                how="left",
+                left_index=True,
+                right_index=True,
+            )
+        else:
+            # sg input
+            ddf = cudf.concat(
+                [
+                    start_list,
+                    batch_id_list,
+                ],
+                axis=1,
+            )
     else:
-        ddf = start_list
+        ddf = start_list.to_frame()
```

Comment:
Do we really care about the index here ? I think not . 

```python3
start_list = start_list.reset_index(drop=True)
batch_id_list = batch_id_list.reset_index(drop=True)

if isinstance(start_list, dask_cudf.Series):
   ddf = dd.concat([start_list, batch_id_list], ignore_unknown_divisions=True, axis=1) 
else:
   ddf = cudf.concat([start_list, batch_id_list], axis =1) 
```

Reply:
If we reset index can we join batch id and start list correctly?

Reply:
And also, I ran into an issue with `dask_cudf.concat` where the name of the series was dropped in one of my first attempts at a solution.  `dask_cudf.merge` doesn't have that problem.

Reply:
I think we should be able to, from the logic you shared , you are merging on index  `( left_index=True, right_index=True)` in dask . 

Edit: Also added `ingore_index=True` to make it more concrete in `cuDF`.

Reply:
ok, let me try this

Reply:
Another thing to try is below: 

```python3
if isinstance(start_list, dask_cudf.Series): 
    start_list.divisions = start_list.compute_current_divisions()
     batch_id_list.divisions = batch_id_list.compute_current_divisions()
    ddf = dd.concat([start_list, batch_id_list], axis=1) 
else:
    ddf = cudf.concat([start_list, batch_id_list], axis =1, ignore_index=True) 
```

Reply:
First solution isn't working, but the second makes sense, I think that should work.  I'm trying it now.

Reply:
2nd solution works when `ignore_index=False`, I'm going to go ahead with that

Reply:
Cool.  Thanks

Reply:
I am confused why it is failing though , any chance you could post a MRE to cuDF (After merging this fix in), getting this fix in is a higher priority.

Reply:
I'll try to get a MRE.  That has been a challenge, though.  What I've been doing is running both the new pytests and my code for processing the MAG dataset.

Reply:
At some point I'd like to turn the MAG dataset examples into tests that get run in the MNMG nightlies.

Reply:
sorry for the delay, I see some tests failing where the column names aren't propagating correctly

Reply:
@VibhuJawa dask_cudf.concat renames the columns to 0 and 1, and when I try to rename them, I get a "cannot rename index" error.

Reply:
Lets file an issue with that detail and just go with your earlier fix. Sorry for the back and forth .


It is unfortunate that we potentially have bugs in such a basic operation in `dask-cudf` and `cuDF`.  Can you add a comment explaining the bugs here to make the code parsable for future.

Reply:
```
ddf = dask_cudf.concat([series1, series2], axis=1)
print(ddf.columns)
```
Expected:
```
Index(['series1_name', 'series2_name'])
```
Actual:
```
Int64Index([0, 1], dtype='int64')
```

Reply:
I'll file the issue with cudf tonight.  I know my fix isn't ideal, but at least for now it's not a huge deal since most of our use cases are passing a cudf (not dask cudf) dataframe.

Reply:
I'm not sure why calling `merge` instead of `concat` before `get_distributed_data` works, but for some reason the bug completely disappears with merge.

Reply:
I can take a look too

Reply:
Thanks for creating an issue .